src/tests/suite/daos_test.h

```diff
@@ -476,8 +476,10 @@ test_rmdir(const char *path, bool force)
 			continue;   /* skips the dots */
 
 		D_ASPRINTF(fullpath, "%s/%s", path, ent->d_name);
-		if (fullpath == NULL)
```

Comment:
I guess if the goal of this patch is to resolve coverity scan issues, then these things could be disregarded for this PR. Observations below.

In the original form the D_GOTO did not set rc = -DER_NOMEM. Not sure if maybe the intention was to do that, in which case it would have been D_GOTO(out, rc = -DER_NOMEM).

In the new patch this is the case and that seems like a good idea. However when the code breaks so that the closedir() can run, the rc that is set here is lost and replaced with whatever closedir() returned. Resolving it is inconvenient because that code may run in the successful case and in the error unwinding case.

Another observation is in some cases when a call that sets errno fails, the function will set rc = daos_errno2der() and in other cases it doesn't

Reply:
I do believe it should have been rc = -DER_NOMEM because otherwise it would amount to a regular goto statement.

I've changed it so it just calls closedir and then jumps to out with rc = -DER_NOMEM instead of going through the normal file close logic at the bottom. This seems closer to what the original was doing while still attempting to close/free the file.